hs_core/models.py

```diff
@@ -1127,11 +1128,17 @@ def delete(self, using=None):
         self.metadata.delete_all_elements()
         self.metadata.delete()
 
+        # delete related access controls records
         access_resource = self.raccess
         UserResourcePrivilege.objects.filter(resource=access_resource).delete()
         GroupResourcePrivilege.objects.filter(resource=access_resource).delete()
         access_resource.delete()
```

Comment:
I will make the FKs non-null as discussed above and then the chaining will definitely work. This is cleaner. The save_labels function can be re-implemented inside the UserLabels object instead of the UserResourceLabels object. This makes more sense.